src/structure.rs

```diff
@@ -136,6 +135,15 @@ impl<'a> Catalog<'a> {
     pub fn output_intents(&mut self) -> TypedArray<'_, Dict> {
         self.insert(Name(b"OutputIntents")).array().typed()
     }
+
+    /// Start writing the `/AF` array to specify the
+    /// associated files of the document. PDF 2.0+.
+    ///
+    /// Each entry in the array is a reference to a [file specification
+    /// dictionary.](writers::FileSpec)
+    pub fn associated_files(&mut self) -> TypedArray<'_, Ref> {
```

Comment:
It seems inconsistent that this requires an array of `Ref`s while `output_intents` takes direct dictionaries. It is a bit unfortunate that pdf-writer forces one or the other, but while it does, we should be consistent.

Reply:
I was looking for an example in `pdf_writer` that allows both, refs and objects. But since I couldn't find one, I decided for Refs since they seem to be more flexible.
For example, I need to reference the same file specification dictionary in the `\AF` and the `\Names` array:
![Screenshot from 2024-08-27 20-44-25](https://github.com/user-attachments/assets/c7fa543a-c846-49a6-b548-d1df60c8c274)
Is that possible, when I put the file specification directory into the AF array?

Reply:
Yeah, no, it wouldn't be possible. Some day, I'll try and rework pdf-writer to allow both always.

Reply:
Ok, I will change it to take the directories for consistency.
Maybe I'll get to experimenting with the API later and see if I can find a solution to support both.

Reply:
In the end it's always possible for the user to just copy-paste the raw impl of the function if they need the escape hatch. It's not nice, but we also sometimes do it in Typst. That's the way it is until we have a solution to the problem.